src/join_variant_summary_with_clinvar_alleles.py

```diff
@@ -0,0 +1,113 @@
+
+    clinvar_alleles = pd.read_csv(clinvar_alleles_table, sep="\t",
+                                  index_col=False, compression="gzip")
+    print "clinvar_alleles raw", clinvar_alleles.shape
+
+    # use lowercase names and replace . with _ in column names:
+    variant_summary = variant_summary.rename(columns=dict(
+        (col, col.lower().replace(".", "_"))
+        for col in variant_summary.columns))
+    # rename first column to allele_id:
+    variant_summary = variant_summary.rename(
+        columns={variant_summary.columns[0]: "allele_id"})
+
+    # extract relevant columns for the correct assembly and
+    # rename clinicalsignificance, reviewstatus:
+    variant_summary = variant_summary[
+        variant_summary['assembly'] == genome_build_id]
+    variant_summary = variant_summary[
+        ['allele_id', 'clinicalsignificance', 'reviewstatus']]
+    variant_summary = variant_summary.rename(
+        columns={'clinicalsignificance': 'clinical_significance',
+                 'reviewstatus': 'review_status'})
+    print "variant_summary after filter", variant_summary.shape
+
+    # remove clinical_significance and review_status from clinvar_alleles:
+    clinvar_alleles = clinvar_alleles.drop(
+        ['clinical_significance', 'review_status'], axis=1)
+
+    # remove rows with multiple allele_ids
+    clinvar_alleles = clinvar_alleles[
+        ~(clinvar_alleles['allele_id'].str.contains(";") == True)]
+    # have to do it this way because allele_id gets a dtype of object, being
+    # a mix of ints and strs; alternate way would be to cast
+    # variant_summary.allele_id to str and let pd.merge handle it
+    clinvar_alleles['allele_id'] = clinvar_alleles.allele_id.astype(int)
+
+    print "clinvar_alleles after filter", clinvar_alleles.shape
+
+    # join the two tables on allele_id:
+    df = pd.merge(clinvar_alleles, variant_summary,
+                  on='allele_id', how='inner')
+    print "merged raw", df.shape
+
+    # map review_status to gold starts:
+    gold_star_map = {
+        'no assertion provided': 0,
+        'no assertion for the individual variant': 0,
+        'no assertion criteria provided': 0,
+        'criteria provided, single submitter': 1,
+        'criteria provided, conflicting interpretations': 1,
+        'criteria provided, multiple submitters, no conflicts': 2,
+        'reviewed by expert panel': 3,
+        'practice guideline': 4
+    }
+    df['gold_stars'] = df.review_status.map(gold_star_map)
+
+    # pathogenic = 1 if at least one submission says pathogenic or likely
+    # pathogenic, 0 otherwise
+    df['pathogenic'] = df['clinical_significance'].str.contains(
+        "pathogenic", case=False)
+    # benign = 1 if at least one submission says benign or likely benign,
+    # 0 otherwise
+    df['benign'] = df['clinical_significance'].str.contains(
+        "benign", case=False)
+    # we have to deal with cases of strings like "probable-non-pathogenic",
```

Comment:
The description of clinical significance used by ClinVar is restricted by a certain set of terms: https://www.ncbi.nlm.nih.gov/clinvar/docs/clinsig/.

The variant_summary file is aggregated from RCV records and it is not likely to use these terms. It could also be checked by looking at the unique values used in the column "clinical_significance".